src/pages/reports/Reports.tsx

```diff
@@ -1,16 +1,108 @@
-            content={
-                <div className="flex flex-col">
-                   
-                </div>
-            } />
-    );
+import React, { useState } from 'react';
+import { Button, Collapse, DatePicker, Form, Table, Row, Col } from 'antd';
+import LayoutCard from '../layouts/LayoutCard';
+import SectionElement from '../components/SectionElement';
+import Strings from '../../utils/Strings';
+import { usePostReportsMutation } from '../../services/reportsService';
+import { handleErrorNotification } from '../../utils/Notifications';
+import { Reports } from '../../data/reports/report';
+const { Panel } = Collapse;
+
+interface Report {
+	name: string;
+	lastname: string;
+	primaryContact: string;
+	appointment: string;
 }
 
-export default Reports;
\ No newline at end of file
+const ReportsComponent: React.FC = () => {
+	const [started_at, setStarted_at] = useState<string | null>(null);
+	const [finished_at, setFinished_at] = useState<string | null>(null);
+	const [tableData, setTableData] = useState<Report[]>([]);
+	const [postReports] = usePostReportsMutation();
+
+	const handleConsultClick = async () => {
+		try {
+			const result = await postReports({ started_at, finished_at }).unwrap();
+			if (Array.isArray(result)) {
+				const mappedData: Report[] = result.map((item: Reports) => ({
+					name: item.patient.name,
+					lastname: item.patient.lastname,
+					primaryContact: item.patient.primaryContact,
+					appointment: item.appointment,
+				}));
+
+				setTableData(mappedData);
+			}
+		} catch (error) {
+			handleErrorNotification(error);
+		}
+	};
+
+	
+	return (
+		<LayoutCard
+		  title={Strings.reports}
+		  isLoading={false}
+		  content={
+			<Collapse>
+			  <Panel header="Visualización de reportes" key="1">
+				<div className="flex justify-between items-center">
+				  <div className="flex">
+					<div>
+					  <SectionElement
+						label={Strings.started_at}
+						icon={<></>}
+						value={``}
+					  />
+					  <Form.Item
+						rules={[{ required: true, message: Strings.requiredField }]}
+					  >
+						<DatePicker
+						  size="large"
+						  style={{ minWidth: 200 }}
+						  onChange={(date, dateString) => setStarted_at(dateString)}
+						/>
+					  </Form.Item>
+					</div>
+	  
+					<div style={{ marginLeft: '20px' }}>
+					  <SectionElement
+						label={Strings.finished_at}
+						icon={<></>}
+						value={``}
+					  />
+					  <Form.Item
+						rules={[{ required: true, message: Strings.requiredField }]}
+					  >
+						<DatePicker
+						  size="large"
+						  style={{ minWidth: 200 }}
+						  onChange={(date, dateString) => setFinished_at(dateString)}
+						/>
+					  </Form.Item>
+					</div>
+				  </div>
+	  
+				  <div>
+					<Button type="primary" onClick={() => handleConsultClick()}>
+					  {Strings.ReportConsult}
+					</Button>
+				  </div>
+				</div>
+	  
+				<Table dataSource={tableData}>
+				  <Table.Column title="Nombre" dataIndex="name" key="name" />
```

Comment:
Please use the constants here for all the strings

Reply:
Okay, I already updated it. I attach an example image
![Captura de pantalla 2024-01-24 230151](https://github.com/cDentalCareGroup/web-cdentalcaregroup/assets/84301741/cc909f9f-0a1d-4479-b96e-ba2aa05c21b8)

Reply:
Thanks

Reply:
In this method 

const handleConsultClick = async () => {
		try {
			const result = await postReports({ started_at, finished_at }).unwrap();
			if (Array.isArray(result)) {

you check if the result its an array but with this cast you should't do this